aws-lc-rs/src/agreement.rs

```diff
@@ -539,65 +548,79 @@ const MAX_AGREEMENT_SECRET_LEN: usize = ECDH_P521_PRIVATE_KEY_LEN;
 
 #[inline]
 #[allow(clippy::needless_pass_by_value)]
-unsafe fn ec_key_ecdh<'a>(
+fn ec_key_ecdh<'a>(
     buffer: &'a mut [u8; MAX_AGREEMENT_SECRET_LEN],
-    priv_ec_key: ConstPointer<EC_KEY>,
+    priv_key: &LcPtr<*mut EVP_PKEY>,
     peer_pub_key_bytes: &[u8],
     nid: i32,
 ) -> Result<&'a [u8], ()> {
-    let ec_group = ec_group_from_nid(nid)?;
-    let pub_key_point = ec_point_from_bytes(&ec_group, peer_pub_key_bytes)?;
-    let peer_ec_key = ec_key_from_public_point(&ec_group, &pub_key_point)?;
+    let ec_group = unsafe { ec_group_from_nid(nid)? };
+    let pub_key_point = unsafe { ec_point_from_bytes(&ec_group, peer_pub_key_bytes) }?;
+    let peer_ec_key = unsafe { ec_key_from_public_point(&ec_group, &pub_key_point) }?;
 
-    let priv_group = ConstPointer::new(EC_KEY_get0_group(*priv_ec_key))?;
-    let priv_nid = EC_GROUP_get_curve_name(*priv_group);
+    let pkey_ctx = LcPtr::new(unsafe { EVP_PKEY_CTX_new(**priv_key, null_mut()) })?;
 
-    let supported_curves = [NID_X9_62_prime256v1, NID_secp384r1, NID_secp521r1];
-    if !supported_curves.contains(&priv_nid as &i32) {
```

Comment:
From my investigation this check was already covered and duplicative which is why I removed it.

`EVP_PKEY_derive` will call `ECDH_compute_shared_secret` which has the following check: https://github.com/aws/aws-lc/blob/c2c8d66d28bd0421fbe8646e9b1b7e4f94e52757/crypto/fipsmodule/ecdh/ecdh.c#L89-L94